File: 백준/Silver/1913. 달팽이/달팽이.java
import java.util.*;
import java.io.*;

public class Main {
	
	static int[][] map;
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		
		int num = Integer.parseInt(br.readLine());
		int find_num = Integer.parseInt(br.readLine());
		
		snail(num);
		
		int x=0, y=0;
		for(int i=0; i<num; i++) {
			for(int j=0; j<num; j++) {
				if(find_num == map[i][j]) {
					y = i+1;
					x = j+1;
				}
				sb.append(map[i][j]+" ");
				}
			sb.append("\n");
			}
			sb.append(y+" "+x);
			System.out.println(sb.toString());
		}
	
	
	public static void snail(int n) {
		
		map= new int[n][n];
		int temp=0;
		int count = n*n;
		
		while(count>0) {
			
			for(int i=temp; i<n; i++) {
				map[i][temp] = count--;
			}
			
			for(int i=temp+1; i<n; i++) {
				map[n-1][i] = count--;
			}
			
			for(int i=n-2; i>=temp; i--) {
				map[i][n-1] = count--;
			}
			
			for(int i=n-2; i>temp; i--) {
				map[temp][i] = count--;
			}
		
		temp++;
		n--;
		}
	}
}

File: 백준/Silver/1913. 달팽이/달팽이.cc
#include <iostream>
#include <algorithm>
#include <map>
#include <string>

using namespace std;
int arr[1001][1001];

void snail(int n) {
	int temp = 0;
	int cnt = n*n;
	while (cnt > 0) {

		for (int i = temp; i < n; i++) {
			arr[i][temp] = cnt--;
		}

		for (int i = temp + 1; i < n; i++) {
			arr[n - 1][i] = cnt--;
		}

		for (int i = n - 2; i >= temp; i--) {
			arr[i][n - 1] = cnt--;
		}

		for (int i = n - 2; i > temp; i--) {
			arr[temp][i] = cnt--;
		}
		temp++;
		n--;
	}
}

int main(void) {
	
	int n, num;
	cin >> n >> num;

	snail(n);

	int x = 0, y = 0;
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			if (arr[i][j] == num) {
				y = i + 1;
				x = j + 1;
			}
			cout << arr[i][j] << ' ';
		}
		cout << "\n";
	}
	cout << y << ' ' << x;
}